build-logic/convention/src/main/kotlin/com/google/samples/apps/nowinandroid/AndroidCompose.kt

```diff
@@ -52,11 +54,24 @@ internal fun Project.configureAndroidCompose(
 
     extensions.configure<ComposeCompilerGradlePluginExtension> {
         if (isPropertyValueIsTrue("enableComposeCompilerReportsAndMetrics")) {
-            metricsDestination = relativeToRootProject("compose-metrics")
-            reportsDestination = relativeToRootProject("compose-reports")
+            metricsDestination = createPerModulePath("compose-metrics")
+            reportsDestination = createPerModulePath("compose-reports")
         }
 
-        stabilityConfigurationFile =
-            rootProject.layout.projectDirectory.file("compose_compiler_config.conf")
+        stabilityConfigurationFiles
+            .add(isolated.rootProject.projectDirectory.file("compose_compiler_config.conf"))
     }
 }
+
+/**
+ * Checking a value of [propertyName] is true.
+ */
+private fun Project.isPropertyValueIsTrue(propertyName: String): Boolean =
+    providers.gradleProperty(propertyName).orNull.toBoolean()
```

Comment:
nit: as said earlier, there is not point of extracting this into a function. It is only used once, and the naming is not meaningful.
Let's simply do this at the call site:

```kotlin
if (providers.gradleProperty("enableComposeCompilerReportsAndMetrics").orNull.toBoolean()) {
```

or extract as a variable if we really want it:

```kotlin
val isComposeCompilerReportsAndMetricsEnabled = providers.gradleProperty("enableComposeCompilerReportsAndMetrics").orNull.toBoolean()
if (isComposeCompilerReportsAndMetricsEnabled) {
```

Reply:
It also looks good, I'm accepting your suggestion.